Src/ILGPU/Backends/OpenCL/CLCodeGenerator.Values.cs

```diff
@@ -398,6 +399,82 @@ public void GenerateCode(LoadFieldAddress value)
             statement.AppendFieldViaPtr(value.FieldSpan.Access);
         }
 
+        /// <summary cref="IBackendCodeGenerator.GenerateCode(AlignTo)"/>
+        public void GenerateCode(AlignTo value)
+        {
+            // Load the base view pointer
+            var source = Load(value.Source);
+            var target = Allocate(value);
+            var alignmentVariable = Load(value.AlignmentInBytes);
+            var arithmeticBasicValueType =
+                value.Source.BasicValueType.GetArithmeticBasicValueType(true);
+
+            // var baseOffset = (int)ptr & (alignmentInBytes - 1);
+            var baseOffset = AllocateType(arithmeticBasicValueType);
+            using (var statement = BeginStatement(baseOffset))
+            {
+                statement.AppendCast(arithmeticBasicValueType);
+                statement.AppendArgument(source);
+                statement.AppendCommand(CLInstructions.GetArithmeticOperation(
+                    BinaryArithmeticKind.And,
+                    isFloat: false,
+                    out bool _));
+                // Optimize for the case in which the alignment is a constant value
+                if (value.TryGetAlignmentConstant(out int alignment))
+                {
+                    statement.AppendConstant(alignment - 1);
+                }
+                else
+                {
+                    statement.AppendCommand('(');
+                    statement.AppendArgument(alignmentVariable);
+                    statement.AppendCommand(CLInstructions.GetArithmeticOperation(
+                        BinaryArithmeticKind.Sub,
+                        isFloat: false,
+                        out bool _));
+                    statement.AppendConstant(1);
+                    statement.AppendCommand(')');
+                }
+            }
+
+            // if (baseOffset == 0) { 0 } else { alignment - baseOffset }
+            var adjustment = AllocateType(arithmeticBasicValueType);
+            using (var selectStatement = BeginStatement(adjustment))
+            {
+                selectStatement.AppendArgument(baseOffset);
+                selectStatement.AppendCommand(" == 0 ? 0 : ");
```

Comment:
ISSUE: The ternary operator may need brackets.
Currently, `baseOffset == 0 ? 0 : alignment - baseOffset` is ambiguous.
Should use `baseOffset == 0 ? 0 : (alignment - baseOffset)`.

Reply:
Nice catch! This will fail to compile on several OpenCL platforms.